pkg/ccl/backupccl/show_test.go

```diff
@@ -467,8 +467,44 @@ func TestShowBackups(t *testing.T) {
 	const full = localFoo + "/full"
 	const remoteInc = localFoo + "/inc"
 
+	t.Run("nonDefault", func(t *testing.T) {
+		// Make an initial backup.
+		fullNonDefault := full + "NonDefault"
+		incNonDefault := remoteInc + "NonDefault"
+		sqlDB.Exec(t, `BACKUP DATABASE data INTO $1`, fullNonDefault)
+
+		// Get base number of files, schemas, and ranges in the backup
+		var oldCount [3]int
+		for i, typ := range []string{"FILES", "SCHEMAS", "RANGES"} {
+			query := fmt.Sprintf(`SELECT count(*) FROM [SHOW BACKUP %s LATEST IN '%s']`, typ, fullNonDefault)
+			count, err := strconv.Atoi(sqlDB.QueryStr(t, query)[0][0])
+			require.NoError(t, err, "error converting original count to integer")
+			oldCount[i] = count
+		}
+
+		// Increase the number of files,schemas, and ranges that will be in the backup chain
+		sqlDB.Exec(t, `CREATE TABLE data.blob (a INT PRIMARY KEY); INSERT INTO data.blob VALUES (0)`)
+		sqlDB.Exec(t, `BACKUP INTO LATEST IN $1`, fullNonDefault)
+		sqlDB.Exec(t, `BACKUP INTO LATEST IN $1 WITH incremental_location=$2`, fullNonDefault, incNonDefault)
+
+		// Show backup should contain more rows as new files/schemas/ranges were
+		// added in the incremental backup
+		for i, typ := range []string{"FILES", "SCHEMAS", "RANGES"} {
+			query := fmt.Sprintf(`SELECT count(*) FROM [SHOW BACKUP %s LATEST IN '%s']`, typ, fullNonDefault)
+			newCount, err := strconv.Atoi(sqlDB.QueryStr(t, query)[0][0])
+			require.NoError(t, err, "error converting new count to integer")
+			require.Greater(t, newCount, oldCount[i])
+
+			queryInc := fmt.Sprintf(`SELECT count(*) FROM [SHOW BACKUP %s LATEST IN '%s' WITH incremental_location='%s']`, typ,
+				fullNonDefault, incNonDefault)
+			newCountInc, err := strconv.Atoi(sqlDB.QueryStr(t, queryInc)[0][0])
+			require.NoError(t, err, "error converting new count to integer")
+			require.Greater(t, newCountInc, oldCount[i])
+		}
+	})
 	// Make an initial backup.
```

Comment:
nit: can we wrap the test below into its own subtest too?

I'd love to see all of this move to datadriven tests as part of https://github.com/cockroachdb/cockroach/issues/77129. The tests aren't doing much apart from issuing SQL queries so it should be straightforward to port. Not asking for it to be done in this PR, but generally hope we can start reaching for DD first.

Reply:
decided to move my new subtest to its own test, as this commit will likely get backported. I've added a todo for myself to make these tests data driven. good callout.